source.go

```diff
@@ -8,11 +8,147 @@ import (
 	"strings"
 
 	"github.com/moby/buildkit/client/llb"
+	"github.com/moby/buildkit/util/bklog"
 	"github.com/moby/buildkit/util/gitutil"
 	"github.com/pkg/errors"
 )
 
-type LLBGetter func(sOpts SourceOpts, opts ...llb.ConstraintsOpt) (llb.State, error)
+type FilterFunc = func(string, []string, []string, ...llb.ConstraintsOpt) llb.StateOption
```

Comment:
In general I'm wondering if we need to be passing around a `FilterFunc`.
If we do, then it should be  passed as an `llb.StateOption` rather than a `func() -> llb.StateOption`

Reply:
I think I see what you're getting at now. There is no need for this to be passed in, it can be applied by the caller. I am changing to this approach